### DISCUSSION TITLE
箱庭ドローンシミュレータをお試し中、ニューヨーク上空を飛んでみたい。

### DISCUSSION THREAD

**Discussion by yamati-kz** [toppers/hakoniwa#184] [Q&A]
@tmori さん、こんにちは。いろいろ教えていただいてありがとうございます。

つまずきながら、なんとかUnityアプリでドローンシミュレータを動かすことができました。
Unity Editorはまだでこれからになります。OSは、Ubuntu 22.04(Native)です。
PS4ではなくPS5コントローラ(DualSense)をUSB接続しました。
QGroundControl とのデータのやり取りが心配でしたが、JoyStick操作で上昇下降や旋回も動作しています。

試している中でマニュアルの記載で少々気になったといいますか、まあわかるでしょうレベルの内容ですが。
cd (change directory)が抜けている、一箇所Directory名が違っている程度で流れに問題があるわけではないです。
Fork後のMainリポジトリの中に手をいれたubuntu22.04_install&operation_PX4編.mdが入ってますので、
もしよかったら使っていただけたらとおもます。
[修正内容ご参考](https://github.com/yamati-kz/hakoniwa-px4sim/pull/1/files) .

徐々にニューヨーク上空にチャレンジしていきますので、ご報告がてらまた書き込みさせてください。

今後ともよろしくおねがいします。

追記 : ついに飛べました!! 大袈裟ですが...  @tmori さんのおかげです。本当に感謝!!
[箱庭ドローンシミュレータでニューヨーク上空を飛べた！ (PS5 Controller DualSense使用)](https://qiita.com/yamati-kz/items/43bf7269dcdf4a898cd9)

**Comment by tmori** [comment]
@yamati-kz さん、箱庭ドローンシミュレータをご利用いただきありがとうございます！
また、ご指摘もありがとうございます。

ご指摘いただいた点含めて、修正検討進めたいと思いますので、
今後ともどうぞよろしくお願いいたします。

**Reply by yamati-kz** [reply]
ありがとうございます。ではPRさせていただきます。

**Comment by yamati-kz** [comment]
@tmori さん、お忙しい中ご返信ありがとうございます！

 ubuntu22.04_install&operation_PX4編.md に [python3-devの追加](https://github.com/yamati-kz/hakoniwa-px4sim/pull/2/files) もご検討くださいますと嬉しいです！
pyhton関連では当たり前かとは思いますが.

もしPullRequestさせていただくとすると、[hakoniwa-px4sim/tree/develop](https://github.com/toppers/hakoniwa-px4sim/tree/develop)でよろしいでしょうか？

**Reply by tmori** [reply]
ありがとうございます！
PR大歓迎です。ご推察の通り、developにPRいただけますと大変ありがたいです。
現状、developは開発用のブランチとして管理しており、品質安定したタイミングでmainにマージする形で運用を進めております。

**Comment by yamati-kz** [comment]
@tmori さん

小姑のようですみません。
ubuntu22.04_install&operation_PX4編.md 0.5版にて試したところ、以下の部分で引っかかりましたので、ご検討いただけたら幸いです。
![image](https://github.com/user-attachments/assets/314f5fc6-d73f-49cd-9ffc-cadf7d177fc9)

**Reply by tmori** [reply]
ご連絡ありがとうございます！
修正しましたので、ご確認くださいませませ。

https://github.com/toppers/hakoniwa-px4sim/commit/811f69a0b7beb47a85045e028ee74670041fd731

**Reply by yamati-kz** [reply]
ご対応ありがとうございました。該当ファイルを落として確認いたしました。

**Comment by yamati-kz** [comment]
@tmori さん　おはようございます。

急ぎませんので少々教えていただければと思います。
新規に自作した ps5-control-lnx.json をデフォルトで読み込ませる便利な仕組みがありますでしょうか？
諸々追いきれておりません、すみません。

PS5コントローラ(DualSense)のindex確認のために [rc_control.py](https://github.com/toppers/hakoniwa-px4sim/blob/main/drone_api/sample/rc_debug/rc_control.py) を使わさせていただいております。

**Reply by tmori** [reply]
> @tmori さん　おはようございます。
> 
> 急ぎませんので少々教えていただければと思います。 新規に自作した ps5-control-lnx.json をデフォルトで読み込ませる便利な仕組みがありますでしょうか？ 諸々追いきれておりません、すみません。
> 
> PS5コントローラ(DualSense)のindex確認のために [rc_control.py](https://github.com/toppers/hakoniwa-px4sim/blob/main/drone_api/sample/rc_debug/rc_control.py) を使わさせていただいております。

デフォルトとしてはなくて、自作いただいたps5-control-lnx.jsonをrc-custom.pyをご利用いただく形かなと思いますが、答えになっておりますでしょうか。

https://github.com/toppers/hakoniwa-px4sim/blob/main/docs/manual/ubuntu22.04_install%26operation_pythonAPI編.md#144-ps4のコントローラでのドローン操作

案としは、jsonファイルのパスを環境変数で設定して、bash script で rc-custom.pyを起動する感じになると使いやすくなるかなと思います。

あと、もしよろしければ、ps5-control-lnx.jsonをPRいただけるととっても嬉しいです！

**Reply by yamati-kz** [reply]
@tmori さん　ご返事ありがとうございます。

まず、こんなイメージでしょうか？
コマンドラインの変更
![image](https://github.com/user-attachments/assets/aab51344-c7aa-4add-b413-7b777b7e6705)
rc-custom.pyの変更
![image](https://github.com/user-attachments/assets/c8bbf1bc-4cb6-4365-a523-0ec50019ca60)

> デフォルトとしてはなくて、自作いただいたps5-control-lnx.jsonをrc-custom.pyをご利用いただく形かなと思いますが、答えになっておりますでしょうか。

はい、わかってきた気がします。
環境変数とbash scriptの件は、ubuntuに不慣れなもので別途ご相談させてください。

> あと、もしよろしければ、ps5-control-lnx.jsonをPRいただけるととっても嬉しいです！

はい、私なりに動作確認が取れましたらps5-control-lnx.jsonをPRさせていただきますね。
index値以外の意味合いがあやふやなところもありますのでご相談させてください。

**Reply by yamati-kz** [reply]
確認させてください。
箱庭ドローンシミュレータでは、下図にようなボタン割り当てでしょうか？
何かで見た時にメモした内容で下図を作成しております。

![image](https://github.com/user-attachments/assets/514bf8e3-4e52-4980-9700-bb0c9d8015be)

それとも、こちらでしょうか？![PS4コントローラ](https://github.com/toppers/hakoniwa-px4sim/blob/main/docs/drone_control/ps4/image_ps4.png)
![image](https://github.com/user-attachments/assets/298b2219-c4af-4168-b561-74fa7b6e214c)

**Reply by tmori** [reply]
> @tmori さん　ご返事ありがとうございます。
> 
> まず、こんなイメージでしょうか？ コマンドラインの変更 ![image](https://private-user-images.githubusercontent.com/135952435/363892770-aab51344-c7aa-4add-b413-7b777b7e6705.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUzOTc4NDUsIm5iZiI6MTcyNTM5NzU0NSwicGF0aCI6Ii8xMzU5NTI0MzUvMzYzODkyNzcwLWFhYjUxMzQ0LWM3YWEtNGFkZC1iNDEzLTdiNzc3YjdlNjcwNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwM1QyMTA1NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wOTQwYWJhYTk4ZGJiYmQ0YjEzZjBmMDIzNTNiY2ZmNTg3ODFiZmJhOTk5ZTRlYjQ1ZmFmZTQ2OTU5MjU2OWU2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Y-s4tLQrfIK4gFvqvZpoKi7ENXlXP5XU5mo-1hRYhjQ) rc-custom.pyの変更 ![image](https://private-user-images.githubusercontent.com/135952435/363898096-c8bbf1bc-4cb6-4365-a523-0ec50019ca60.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUzOTc4NDUsIm5iZiI6MTcyNTM5NzU0NSwicGF0aCI6Ii8xMzU5NTI0MzUvMzYzODk4MDk2LWM4YmJmMWJjLTRjYjYtNDM2NS1hNTIzLTBlYzUwMDE5Y2E2MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwM1QyMTA1NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01ZTgwMjlmZWE0NTc4NmVmOGE4MmE5OTYzYjUyM2UxZWJhZDFiMWExMDVkODM4MzE3NmEyOTU1NzIzNjM0ZjYwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ubnq75m5IcEZD8HOZtJOwnRXPpA1ARyS4MHEUhrBWq4)
> 
> > デフォルトとしてはなくて、自作いただいたps5-control-lnx.jsonをrc-custom.pyをご利用いただく形かなと思いますが、答えになっておりますでしょうか。
> 
> はい、わかってきた気がします。 環境変数とbash scriptの件は、ubuntuに不慣れなもので別途ご相談させてください。
> 
> > あと、もしよろしければ、ps5-control-lnx.jsonをPRいただけるととっても嬉しいです！
> 
> はい、私なりに動作確認が取れましたらps5-control-lnx.jsonをPRさせていただきますね。 index値以外の意味合いがあやふやなところもありますのでご相談させてください。

修正してみましたので、お手数ですが、ご確認いただけますか。

https://github.com/toppers/hakoniwa-px4sim/commit/df6a68720c16280c5dd6f2eb6650aea669c4e51e

.bashrcなどで、RC_CONFIG_PATH を定義すれば使えると思います。

例：直接エクスポートする場合
```
export RC_CONFIG_PATH=rc_config/ps5-control-lnx.json
python rc-custom.py ../../../hakoniwa-unity-drone-model/custom.json
```

**Reply by tmori** [reply]
> 確認させてください。 箱庭ドローンシミュレータでは、下図にようなボタン割り当てでしょうか？ 何かで見た時にメモした内容で下図を作成しております。

箱庭ドローンシミュレータでは、下記のボタン配置になりますね。

> それとも、こちらでしょうか？![PS4コントローラ](https://github.com/toppers/hakoniwa-px4sim/blob/main/docs/drone_control/ps4/image_ps4.png) ![image](https://private-user-images.githubusercontent.com/135952435/364116661-298b2219-c4af-4168-b561-74fa7b6e214c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUzOTg2MzMsIm5iZiI6MTcyNTM5ODMzMywicGF0aCI6Ii8xMzU5NTI0MzUvMzY0MTE2NjYxLTI5OGIyMjE5LWM0YWYtNDE2OC1iNTYxLTc0ZmE3YjZlMjE0Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwM1QyMTE4NTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MWViNWE4MWYzODc4Y2I3MjZhNWRmNDY1ZmQ1OGQ4ZWJmZWNlOTc1ZTYyYTA2ZWYwZjk3NDFkY2E5MWJjZTExJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Jn6-ErLFGysFfh1wn_9cG1erArytY93uSG9eCu3BSnY)


ただし、カメラの上下の説明がまだ入れてなかったです。。
ご提示いただいた下図の認識であっております。

<img width="122" alt="スクリーンショット 2024-09-04 6 21 36" src="https://github.com/user-attachments/assets/53a513af-338f-484f-b743-66b0c1e8a30b">

**Reply by yamati-kz** [reply]
素早いご対応ありがとうございます。試してみます。

**Reply by yamati-kz** [reply]
> ただし、カメラの上下の説明がまだ入れてなかったです。。
> ご提示いただいた下図の認識であっております。

Generalではなく、Q&Aの方が良いでしょうか？ 新規自作PS5用Jsonファイル(ps5-control-lnx.json)編集に関してなのですが？
どうすれば、下の画像と合わせられるのかご教示いただけると嬉しいです。。
![image](https://github.com/user-attachments/assets/d21cc279-b894-472f-b08f-aa43f595cb2c) 

下のは[hakoniwa-px4sim/drone_api/sample/rc_debug/rc_control.py](https://github.com/toppers/hakoniwa-px4sim/blob/develop/drone_api/sample/rc_debug/rc_control.py)の上図の部分を取り込むところかなと想像しております。
```
       if e.type == pygame.locals.JOYHATMOTION:
            if e.value[0] == 0:
                if e.value[1] == 1:
                    print("十字キー：上")
                elif e.value[1] == 0:
                    print("十字キー：ー")
                else:
                    print("十字キー：下")
            elif e.value[0] == 1:
                if e.value[1] == 0:
                    print("十字キー：右")
                elif e.value[1] == 1:
                    print("十字キー：右上")
                else:
                    print("十字キー：右下")
            else:
                if e.value[1] == 0:
                    print("十字キー：左")
                elif e.value[1] == 1:
                    print("十字キー：左上")
                else:
                    print("十字キー：左下")
```

この部分をps5-control-lnx.jsonにどうのように落とし込むと十字キーを使ってカメラのアップダウンができるようになるのか？
といったところを知りたいです。

[hori4mini-control-lnx.json](https://github.com/toppers/hakoniwa-px4sim/blob/develop/drone_api/sample/rc_config/hori4mini-control-lnx.json)のように設定すれば図とは割当が違いますが、第一弾のお試しはできるのかなとは思っております。

**Reply by tmori** [reply]
すみません、自分は、rc_control.pyはよくわかってないのですが、ポイントとしては、
PS5コントローラのキー割当てがどうなっているかを調べることができるかどうかだと思っております。

PS5コントローラの上ボタンと下ボタンを押下すると、pythonライブラリが`event.button`にインデックス番号を詰めてきます。

そのインデックス番号を jsonファイルに書いておくこということですね。
下図は、PS4コントローラ（Linux版）のモノです。11と12が反応していたので、その値を埋めているという感じですね。

```
        "CameraMoveUp": {
            "index": 11,
            "type": "switch",
            "off": "up",
            "on": "down"
        },
        "CameraMoveDown": {
            "index": 12,
            "type": "switch",
            "off": "up",
            "on": "down"
        }
```

**Reply by yamati-kz** [reply]
@tmori さん
ご対応いただきました https://github.com/toppers/hakoniwa-px4sim/commit/df6a68720c16280c5dd6f2eb6650aea669c4e51e ですが、環境変数を設定する方法でps5-control-lnx.jsonを読んで問題なく動いています。下図です。
![image](https://github.com/user-attachments/assets/8138025b-1b47-4ed6-96f5-50ab4f4c6b39)

そして。
ps5-control-lnx.jsonに十字キー(カメラ上下動)を割り当てる件ですが、現状では十字キーに割り当てできません。
buildko89さんにみていただいたところLinux OSではWindowsと違って十字キーの取り込み方が違うためとのことです。

ps5-control-lnx.jsonですが中身はps4-control-lnx.jsonと全く同じですが、実機で動作確認がとれましたので後ほど[#320](https://github.com/toppers/hakoniwa-px4sim/pull/320)にて PRさせていただきました。

いろいろとありがとうございました。

**Reply by tmori** [reply]
ありがとうございました！とっても嬉しいです！！

**Comment by buildko89** [comment]
@yamati-kz さん
@tmori さんのアドバスであってます。

十字キーの上下でも、LRのボタンでも良いのですが、rc_control.pyで取れた値をindexに設定してみてください。
![rc_control](https://github.com/user-attachments/assets/a68caf8d-f3aa-4082-a625-7d249c4a75f3)

**Reply by yamati-kz** [reply]
@buildko89 さん
コメントありがとうございます。
十字キーを押したときのrc_control.pyでの取れかたが違うようですね。他の方々と違うのはなぜ？
お試しでありますしコントローラのボタン割り当てに拘らなくても json ファイルでカスタマイズ！

以下、rc_control.pyでの十字キーを上下左右、押下した際の結果です。
PS5コントローラ SONY純正DualSense使用時、下図。
![Screenshot from 2024-09-07 04-40-25](https://github.com/user-attachments/assets/7ebc290a-3b7e-4e5c-ac4c-72aaa1f98d16)

PS4コントローラ SONY純正DUALSHOCK4使用時、下図。
![Screenshot from 2024-09-07 04-59-31](https://github.com/user-attachments/assets/d82ee833-6606-4f64-87a7-6260a8f9f0fa)

テスト環境ですが、最低限必要なものだけインストール。Visuall Studio Code よりrc_control.pyを実行しています。
ubuntu 22.04.4 LTS : Native, Clean install (ISOファイルより)
sudo apt install gcc g++ make cmake git jq libgtest-dev net-tools python3-dev  python3-pip
pip3 install pygame

![GW0vApza8AA-FTr](https://github.com/user-attachments/assets/64ea9872-ed5a-4ee7-81c6-bb11a61f1fc6)

**Reply by buildko89** [reply]
@yamati-kz さん
確かに、上下ボタン押したときの挙動が違う見たいです。手元にPS4/PS5のコントローラがないので、お手数ですが、以下の手順
確認頂けると良いかと思います。
rc_control.pyは、お試しで作ったので、値が取れない場合があるかもしれません。

/dev/input/jsxは見えているでしょうか？例えばですが...

$ ls -l /dev/input/js0
/dev/input/js0
と見えているなら、値が取れるはずです…

もし、見えているようでしたら、以下の手順試して頂けると良いかと思います。
$ sudo apt install joystick
$ jstest /dev/input/js0

以下参考リンクです。
[PS4コントローラを使う](https://kanpapa.com/today/2022/09/ros2-joy-ps4-dualshock.html)

**Reply by yamati-kz** [reply]
@buildko89 さん 
$ jstest /dev/input/js0 を試してみました。
SONY純正PS4/SONY純正PS5コントローラでは同じ動きをし、十字キーの上下左右はButtonではなくAxes 7とAxes 6が変化します。

・十字キー、上を押下したとき(純正PS5) 
![Screenshot from 2024-09-07 12-43-01](https://github.com/user-attachments/assets/5c50818f-a0f0-4b24-ad91-a703e77f21aa)

・十字キー、下を押下したとき(純正PS5)
 ![Screenshot from 2024-09-07 12-43-20](https://github.com/user-attachments/assets/8e340c39-375e-4c19-84f0-06e0486f5edb)

・十字キー、左を押下したとき(純正PS5)
![Screenshot from 2024-09-07 12-44-18](https://github.com/user-attachments/assets/122e979d-7bb1-45d7-b556-5c7495974f81)

・十字キー、右を押下したとき(純正PS5)
![Screenshot from 2024-09-07 13-14-50](https://github.com/user-attachments/assets/11d19c1d-f9ec-4c25-95fc-531a2208cce2)

テスト環境ですが、最小限の状態にしております。
ubuntu 22.04.4 LTS : Native, Clean install (ISOファイルより)
$ sudo apt install joystick

**Reply by buildko89** [reply]
@yamati-kz さん
私の方での確認ミスでした。すみません。
OSによって、値の取れ方が変わるので、ボタンで認識したのはWindowsでの確認でした。
Linuxでやると、同じ現象になることが分かりました。

pygameの値の取れ方が違うようで、Linuxの場合は、pygame.locals.JOYHATMOTIONで値を取るようです。
一方、Windowsの場合は、pygame.locals.JOYBUTTONDOWN/pygame.locals.JOYBUTTONUPで値が取れているようです。

ただ、rc-custom.pyの仕様では、pygame.locals.JOYHATMOTIONでの対応ができていないので、対応をするしかなさそうです。
現状は、pygame.locals.JOYBUTTONDOWN/pygame.locals.JOYBUTTONUPで対応できるボタンをカメラの上下操作に割り当て
て対応ください。

![rc_control2](https://github.com/user-attachments/assets/9c4e6ea9-ce5c-4bc4-98f0-cf56fe37c64f)

**Reply by yamati-kz** [reply]
@buildko89 さん
調査いただきましてありがとうございます。

> 現状は、pygame.locals.JOYBUTTONDOWN/pygame.locals.JOYBUTTONUPで対応できるボタンをカメラの上下操作に割り当てて対応ください。

はい、試してみます。お教えいただいてありがとうございます。
PS5(DualSense)では、L1 ボタンとR1ボタンに割付ようかと思います。
ご参考 : [DualSense Wireless Controller Instruction Manual](https://www.playstation.com/content/dam/global_pdc/en/corporate/support/manuals/accessories/ps5-accessories/dualsense-wireless-controller-cfi-zct1w/CFI-ZCT1W_Wireless_Controller_Instruction_Man_EN_MEA.pdf)

**Reply by buildko89** [reply]
ご対応ありがとうございます！

**Comment by buildko89** [comment]
@yamati-kz さん
Ubuntuのマニュアルの修正にご協力頂きありがとうございます。
カメラの制御部分は、マニュアルに反映させておきます。

**Reply by yamati-kz** [reply]
@buildko89 さん
いえ、重箱の隅をつついてすみません。
今後も何かの形でお手伝いさせていただければと存じます。

**Reply by yamati-kz** [reply]
ubuntu22.04_install&operation_pythonAPI編.mdの一箇所のみ修正しました, 
[#321](https://github.com/toppers/hakoniwa-px4sim/pull/321)にてご確認いただければと思います。

**Reply by buildko89** [reply]
ありがとうございます。確認します。

**Comment by yamati-kz** [comment]
@tmori さん　すみません、こちらに失礼します。

Unityエディタのコンソールに出ているエラーを取りたいのですが、どうすれば良いでしょうか？
![image](https://github.com/user-attachments/assets/7e41aba9-1f12-49d5-8969-aa074ad2ffee)

[Hakoniwa-drone-unity](https://www.docswell.com/s/kanetugu2015/KXYR8Y-2024-05-19-160903#)をみながら
com.unity.nuget.newtonsoft-jsonとNO_USE_GRPCは追加してます。

**Reply by yamati-kz** [reply]
@tmori さん　すみません、私のミスです。

以下コマンドの最後のbash install.bshを実行してませんでした。 
```
$ cd ~/work
$ git clone --recursive https://github.com/toppers/hakoniwa-unity-drone-model.git
$ cd hakoniwa-unity-drone-model/
$ bash install.bash
```
ひとまず赤いのは消えて、下図が現状です。
![image](https://github.com/user-attachments/assets/600c7ffe-bc47-4192-ac4b-607882dece12)

**Reply by yamati-kz** [reply]
@tmori さん　こんにちは。おてすきのときに教えていただけたら嬉しいです。
ニューヨークのアセットを読み込み済みで試しているのですが、どうもうまくいきません。
[ubuntu22.04_install&operation_PX4編.md](https://github.com/yamati-kz/hakoniwa-px4sim/blob/main/docs/manual/ubuntu22.04_install%26operation_PX4%E7%B7%A8.md) の手順でUnityアプリの代わりにUnity Editorを使う流れで良いでしょうか？
Core i7 12th(ADL GT2), 32G, SSD:500Gにubuntu 22..04.5をクリーンインストールしたNative環境です。
```
# Terminal-1
# 1.4.1. PX4の起動
$ cd ~/work/hakoniwa-px4sim/px4/PX4-Autopilot
$ bash ../sim/simstart.bash
```
```
# Terminal-2
# 1.4.2. 箱庭コア機能の起動
$ cd ~/work/hakoniwa-px4sim/hakoniwa
$ bash run.bash
```
```
# 1.4.3. Unityアプリの起動 -> Unity Editorを起動
# STARTボタンが押せない (Unity Editor内の右三角をクリックし緑色に変わった)
```
```
# Terminal-3
# 1.5.3. QGCとPX4の動作確認
$ cd ~/work/qgc
$ ./QGroundControl.AppImage
```

下図は、Unity Editorコンソールに赤ビックリマークが多数でているものです。
![unityeditorj](https://github.com/user-attachments/assets/8c1db05d-a5a5-417e-9e52-cd77eb61e80a)

 QGCの状態がDisconnectで状態です。Unityアプリを使ったときは、問題なく操作できます。

**Reply by tmori** [reply]
Unityエディタで、`Generate`は実施されてますでしょうか？

エラーログに、core_config.jsonが見つからないと言われておりますので、箱庭のコンフィグファイルが生成されていないような気がしました。

**Reply by yamati-kz** [reply]
@tmori さん　ニューヨークを飛べました、ありがとうございます！！
Generateしておりませんでした。恥ずかしながら諸々の資料を読み切れてませんでした。
いろいろ試したことをのちほどまとめてみます。

![GX-aMVWaUAE2Opx](https://github.com/user-attachments/assets/9afe9cd0-e1f8-4dd4-a77b-f6b1caa877f8)

**Reply by tmori** [reply]
素晴らしい！

**Reply by yamati-kz** [reply]
一番最初のスレッドにニューヨーク絡みのqiita記事を追記しておきました。
ほんとうにありがとうございました。
